clr/commands.py

```diff
@@ -1,39 +1,47 @@
 from __future__ import print_function
 from importlib import import_module
-from builtins import zip
-from builtins import object
 from dataclasses import dataclass
 import inspect
+from inspect import Signature
 import sys
 import textwrap
 import types
 import difflib
+import shelve
+import os
+from collections import namedtuple
 
 import clr.config
 
 # Sentinal for args get in get_command_spec to indicate there is no default.
-NO_DEFAULT = object()
+# Because we are pickling command specs for clr cache use a random int
+# and check for equality rather than object identity.
+# TODO(michael.cusack): Move command spec to inspect.Signature and remove this.
+NO_DEFAULT = 4194921784511160246
+
+# Sorted list of command namespace keys.
+NAMESPACE_KEYS = sorted(clr.config.commands().keys() | {'system'})
 
-NAMESPACE_KEYS = clr.config.commands().keys() | {'system'}
 # Load lazily namespace modules as needed. Some have expensive/occasionally
 # failing initialization.
 __namespaces = {}
 
-def get_namespaces():
-    # Fill namespace cache
-    for key in NAMESPACE_KEYS: get_namespace(key)
-    return __namespaces
-
 def _load_namespace(key):
+    """Imports the module specified by the given key."""
     if key == 'system':
-        return System()
-
-    module_path = clr.config.commands()[key]
-    try:
-        module = import_module(module_path)
-        return module.COMMANDS
-    except Exception as e:
-        return ErrorLoadingNamespace(key, e)
+        instance = System()
+    else:
+        module_path = clr.config.commands()[key]
+        try:
+            module = import_module(module_path)
+            instance = module.COMMANDS
```

Comment:
Do we expect accessing `module.COMMANDS` to raise an exception of any type? If not, can we move this out of the `try` block and below the `except` line?

It's nice for `try` blocks to have as little code as possible which we expect might raise an exception.